pymc3/distributions/multivariate.py

```diff
@@ -2089,3 +2090,119 @@ def logp(self, value):
 
     def _distr_parameters_for_repr(self):
         return ["mu"]
+
+
+class CAR(Continuous):
+    R"""
+    Likelihood for a conditional autoregression. This is a special case of the
+    multivariate normal with an adjacency-structured covariance matrix.
+
+    .. math::
+
+       f(x \mid W, \alpha, \tau) =
+           \frac{|T|^{1/2}}{(2\pi)^{k/2}}
+           \exp\left\{ -\frac{1}{2} (x-\mu)^{\prime} T^{-1} (x-\mu) \right\}
+
+    where :math:`T = (\tau D(I-\alpha W))^{-1}` and :math:`D = diag(\sum_i W_{ij})`.
+
+    ========  ==========================
+    Support   :math:`x \in \mathbb{R}^k`
+    Mean      :math:`\mu`
+    Variance  :math:`(\tau D(I-\alpha W))^{-1}`
+    ========  ==========================
+
+    Parameters
+    ----------
+    mu: array
+        Mean vector
+    W: matrix
+        Symmetric adjacency matrix of 1s and 0s indicating
+        adjacency between elements.
+    alpha: float or array
+        Autoregression parameter. Values closer to 0 indicate weaker
+        correlation and values closer to 1 indicate higher autocorrelation.
+    tau: float or array
+        Precision variables controlling the scale of the underlying normal variates.
+    sparse: bool, default=False
+        Determines whether or not sparse computations are used
+
+    References
+    ----------
+    ..  Jin, X., Carlin, B., Banerjee, S.
+        "Generalized Hierarchical Multivariate CAR Models for Areal Data"
+        Biometrics, Vol. 61, No. 4 (Dec., 2005), pp. 950-961
+    """
+
+    def __init__(self, mu, W, alpha, tau, sparse=False, *args, **kwargs):
```

Comment:
How does the distribution cope with aesara / random parameters? I ask because I see some scipy and numpy calls during init that could be problematic.

Reply:
For mu, alpha, and tau, it immediately casts these to aesara tensors and assumes they will be tensors afterwards. The only parameter that's assumed to be a Numpy array is the matrix W. There are a few new papers coming out which allow for treating it as a random variable ([see here](https://pdf.sciencedirectassets.com/280960/1-s2.0-S2211675319X00032/1-s2.0-S221167531830143X/am.pdf?X-Amz-Security-Token=IQoJb3JpZ2luX2VjEAcaCXVzLWVhc3QtMSJHMEUCIB0jNZvI7eW8InHgQ7UzM7lOBpboVoBlUA7IG9dCigbOAiEAliXepArWaA5%2FF02zgTl65dsp2NcYy5jAD0kP8bIqh2MqtAMIMBADGgwwNTkwMDM1NDY4NjUiDNK0i1q%2FQIwYZNwetiqRA22naLXLbFDOREGkmGN1au77LJ1LN%2B09kuB3QWzpYiIjcJGZY8QBPa5lB%2BWFU4Vcd3RzMj1cOfYvijaQgUajSWHoj3WHbxWWNXNrSOPvo2KEdVgXgnW%2BsIMaM788nYIzjpq1Ob95JQJ0a7eah%2BMj1unRtBZ6HoSHyGBsegMcMrhlSm3vl5vGO3S1yPO53OGQGK8UiOQ2xH%2BhHBeEM9IZx4kbYj4SohNmAa5ynbsVPxHadF5%2FC2fdQV0Qij9v5W6HmmGJVQP2PZIH9mStzFwnpy73sbJLv%2FzVmntU%2BuWVNJqroEbbJPkKahupvAb%2FbGnfVdEaxv%2B1mP3NNVazcfZa5QEMBtFEse4gNpCscetDSSbOE8ZYJeZDZK4vGiU3iLo9UxFBwG%2FKRCVR%2BKY8xZfXUN08Fay%2F2x7Rs7UirFJ29QN5eYO5YOOmw%2BRXz9bBiEAogIXG4%2Fyfh3WixdHdeuQVP6yi4qlf97aVfLv%2FWVV%2BlqUJc17fMyGSzsML8TDUqRXSl18zeGbgbLntOF8BBtwBYoEQMIvSk4IGOusBNCcKP4SjxnxGIXlLv3mQtx9EXB1WmAfVABQ7Ay6mwKALGEnY28EGkwicISBsWbiDrcVNKlpZkmp2K2hcEnzQ6CI1dbd9L%2BuMMTVnmvXfYkMn4b%2BBgf6Ru0oHMN6%2FoiRdw3cm2UxilviTMTIYh25RDnUgJBDUaepaKAF50HjXeZAzI%2B8%2BugOr4SkSSpgYzcKzSL4STETANLYup3SGqL2xOeWjyd%2BcVwYOKOr0YF7IfIaHhDrx%2BRBptV6zhrTTjtkpFa1kCmNXtQm4G%2FTKoH1jPrNYpO6mvCjs6oDhpCHj2jY1xKFSTSVgNS4vEg%3D%3D&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20210307T153755Z&X-Amz-SignedHeaders=host&X-Amz-Expires=300&X-Amz-Credential=ASIAQ3PHCVTYQE2GSSPQ%2F20210307%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=c2a094b5849d9f350445385355b6d6317d56828d7770166a8c7e5b8638fd1c2a&hash=63c74b618dfd5f33a951938d261a0fb1dd52e5b80ae334dd8925bdb83f33a58d&host=68042c943591013ac2b2430a89b270f6af2c76d8dfd086a07176afe7c76c2c61&pii=S221167531830143X&tid=pdf-4a0f12b0-07f7-4177-a662-318e46e29e40&sid=5fc9372a961a634a665b4b80033991270e1egxrqa&type=client)) but the vast majority of use cases treat W as fixed.

I will note that I'm not really sure how the `self.mean` or `self.mode` attributes are used in other parts of the code base, so I just copied the pattern from other distributions based on the normal to cast mu and set the attributes to be equal to the tensor, as opposed to a Numpy array or whatever else was passed in to `__init__`.

Reply:
Maybe you could mention that w must be a numpy array?

The mean / median / mode etc are used to choose an appropriate default testval. If they are equivalent it's fine to do as you did, I think. Otherwise one of them would be sufficient

Reply:
Fixed; since this is a special case of the MVN, the mean, mode, median are all the same as you suggest.